From 6f498206565c6836967250c08359aa633a0a108f Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Wed, 6 Mar 2019 14:08:34 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3308: use real digital number
 describe pinmux instead of RK_FUNC_n

This fix the existing compile error:
Error: arch/arm64/boot/dts/rockchip/rk3308.dtsi:1765.12-13 syntax error
FATAL ERROR: Unable to parse input tree

And also from the upstrem[0][1], some people don't like the
pointless MACRO RK_FUNC_n.

All the modifications done with sed:

sed -i -e 's/RK_FUNC_GPIO/0/' arch/arm64/boot/dts/rockchip/rk3308*
sed -i -e 's/RK_FUNC_//' arch/arm64/boot/dts/rockchip/rk3308*

[0] https://patchwork.kernel.org/patch/9625173/
[1] https://patchwork.kernel.org/patch/9626883/

Change-Id: Icb7c36fb6bd152628ddb911fc221f65e105e5839
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 .../boot/dts/rockchip/rk3308-ai-va-v10.dts    |   8 +-
 .../boot/dts/rockchip/rk3308-evb-ext-v10.dtsi |   6 +-
 .../boot/dts/rockchip/rk3308-evb-v10.dtsi     |   6 +-
 .../boot/dts/rockchip/rk3308-evb-v11.dtsi     |   8 +-
 arch/arm64/boot/dts/rockchip/rk3308.dtsi      | 184 +++++++++---------
 .../boot/dts/rockchip/rk3308b-evb-v10.dtsi    |   8 +-
 6 files changed, 110 insertions(+), 110 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3308-ai-va-v10.dts b/arch/arm64/boot/dts/rockchip/rk3308-ai-va-v10.dts
index ab6e8a8d1535..7c6390dbedd7 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-ai-va-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-ai-va-v10.dts
@@ -604,19 +604,19 @@
 
 	buttons {
 		mic_mute: mic-mute {
-			rockchip,pins = <0 RK_PC0 RK_FUNC_GPIO &pcfg_pull_up>;
+			rockchip,pins = <0 RK_PC0 0 &pcfg_pull_up>;
 		};
 	};
 	rotary {
 		rotary_gpio: rotary-gpio {
 			rockchip,pins =
-				<2 RK_PB3 RK_FUNC_GPIO &pcfg_pull_none>,
-				<2 RK_PB4 RK_FUNC_GPIO &pcfg_pull_none>;
+				<2 RK_PB3 0 &pcfg_pull_none>,
+				<2 RK_PB4 0 &pcfg_pull_none>;
 		};
 	};
 	sdio-pwrseq {
 		wifi_enable_h: wifi-enable-h {
-			rockchip,pins = <0 RK_PA2 RK_FUNC_GPIO &pcfg_pull_none>;
+			rockchip,pins = <0 RK_PA2 0 &pcfg_pull_none>;
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-ext-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3308-evb-ext-v10.dtsi
index 1c0e66384c16..fa4a36b5cb18 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-ext-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-ext-v10.dtsi
@@ -196,11 +196,11 @@
 		spi_init_cmd: spi-init-cmd {
 			rockchip,pins =
 				/* spi sdi */
-				<1 RK_PC7 RK_FUNC_GPIO &pcfg_pull_none>,
+				<1 RK_PC7 0 &pcfg_pull_none>,
 				/* spi scl */
-				<1 RK_PD0 RK_FUNC_GPIO &pcfg_pull_none>,
+				<1 RK_PD0 0 &pcfg_pull_none>,
 				/* spi cs */
-				<1 RK_PD1 RK_FUNC_GPIO &pcfg_pull_none>;
+				<1 RK_PD1 0 &pcfg_pull_none>;
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
index 941fa1229822..42e7865be76c 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
@@ -721,20 +721,20 @@
 
 	buttons {
 		pwr_key: pwr-key {
-			rockchip,pins = <0 RK_PA6 RK_FUNC_GPIO &pcfg_pull_up>;
+			rockchip,pins = <0 RK_PA6 0 &pcfg_pull_up>;
 		};
 	};
 
 	usb {
 		usb_drv: usb-drv {
 			rockchip,pins =
-				<0 RK_PC5 RK_FUNC_GPIO &pcfg_pull_none>;
+				<0 RK_PC5 0 &pcfg_pull_none>;
 		};
 	};
 
 	sdio-pwrseq {
 		wifi_enable_h: wifi-enable-h {
-			rockchip,pins = <0 RK_PA2 RK_FUNC_GPIO &pcfg_pull_none>;
+			rockchip,pins = <0 RK_PA2 0 &pcfg_pull_none>;
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-v11.dtsi b/arch/arm64/boot/dts/rockchip/rk3308-evb-v11.dtsi
index 2c6a177f27b2..3cc5ddfe925a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-v11.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-v11.dtsi
@@ -771,25 +771,25 @@
 
 	buttons {
 		pwr_key: pwr-key {
-			rockchip,pins = <0 RK_PA6 RK_FUNC_GPIO &pcfg_pull_up>;
+			rockchip,pins = <0 RK_PA6 0 &pcfg_pull_up>;
 		};
 	};
 
 	usb {
 		usb_drv: usb-drv {
-			rockchip,pins = <0 RK_PC5 RK_FUNC_GPIO &pcfg_pull_none>;
+			rockchip,pins = <0 RK_PC5 0 &pcfg_pull_none>;
 		};
 	};
 
 	sdio-pwrseq {
 		wifi_enable_h: wifi-enable-h {
-			rockchip,pins = <0 RK_PA2 RK_FUNC_GPIO &pcfg_pull_none>;
+			rockchip,pins = <0 RK_PA2 0 &pcfg_pull_none>;
 		};
 	};
 
 	wireless-wlan {
 		wifi_wake_host: wifi-wake-host {
-			rockchip,pins = <0 RK_PA0 RK_FUNC_GPIO &pcfg_pull_up>;
+			rockchip,pins = <0 RK_PA0 0 &pcfg_pull_up>;
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3308.dtsi b/arch/arm64/boot/dts/rockchip/rk3308.dtsi
index 2333508e29ea..24483dde6e98 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308.dtsi
@@ -1515,198 +1515,198 @@
 		i2s_2ch_0 {
 			i2s_2ch_0_mclk: i2s-2ch-0-mclk {
 				rockchip,pins =
-					<4 RK_PB4 RK_FUNC_1 &pcfg_pull_none>;
+					<4 RK_PB4 1 &pcfg_pull_none>;
 			};
 
 			i2s_2ch_0_sclk: i2s-2ch-0-sclk {
 				rockchip,pins =
-					<4 RK_PB5 RK_FUNC_1 &pcfg_pull_none>;
+					<4 RK_PB5 1 &pcfg_pull_none>;
 			};
 
 			i2s_2ch_0_lrck: i2s-2ch-0-lrck {
 				rockchip,pins =
-					<4 RK_PB6 RK_FUNC_1 &pcfg_pull_none>;
+					<4 RK_PB6 1 &pcfg_pull_none>;
 			};
 
 			i2s_2ch_0_sdo: i2s-2ch-0-sdo {
 				rockchip,pins =
-					<4 RK_PB7 RK_FUNC_1 &pcfg_pull_none>;
+					<4 RK_PB7 1 &pcfg_pull_none>;
 			};
 
 			i2s_2ch_0_sdi: i2s-2ch-0-sdi {
 				rockchip,pins =
-					<4 RK_PC0 RK_FUNC_1 &pcfg_pull_none>;
+					<4 RK_PC0 1 &pcfg_pull_none>;
 			};
 		};
 
 		i2s_8ch_0 {
 			i2s_8ch_0_mclk: i2s-8ch-0-mclk {
 				rockchip,pins =
-					<2 RK_PA4 RK_FUNC_1 &pcfg_pull_none>;
+					<2 RK_PA4 1 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_0_sclktx: i2s-8ch-0-sclktx {
 				rockchip,pins =
-					<2 RK_PA5 RK_FUNC_1 &pcfg_pull_none>;
+					<2 RK_PA5 1 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_0_sclkrx: i2s-8ch-0-sclkrx {
 				rockchip,pins =
-					<2 RK_PA6 RK_FUNC_1 &pcfg_pull_none>;
+					<2 RK_PA6 1 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_0_lrcktx: i2s-8ch-0-lrcktx {
 				rockchip,pins =
-					<2 RK_PA7 RK_FUNC_1 &pcfg_pull_none>;
+					<2 RK_PA7 1 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_0_lrckrx: i2s-8ch-0-lrckrx {
 				rockchip,pins =
-					<2 RK_PB0 RK_FUNC_1 &pcfg_pull_none>;
+					<2 RK_PB0 1 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_0_sdo0: i2s-8ch-0-sdo0 {
 				rockchip,pins =
-					<2 RK_PB1 RK_FUNC_1 &pcfg_pull_none>;
+					<2 RK_PB1 1 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_0_sdo1: i2s-8ch-0-sdo1 {
 				rockchip,pins =
-					<2 RK_PB2 RK_FUNC_1 &pcfg_pull_none>;
+					<2 RK_PB2 1 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_0_sdo2: i2s-8ch-0-sdo2 {
 				rockchip,pins =
-					<2 RK_PB3 RK_FUNC_1 &pcfg_pull_none>;
+					<2 RK_PB3 1 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_0_sdo3: i2s-8ch-0-sdo3 {
 				rockchip,pins =
-					<2 RK_PB4 RK_FUNC_1 &pcfg_pull_none>;
+					<2 RK_PB4 1 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_0_sdi0: i2s-8ch-0-sdi0 {
 				rockchip,pins =
-					<2 RK_PB5 RK_FUNC_1 &pcfg_pull_none>;
+					<2 RK_PB5 1 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_0_sdi1: i2s-8ch-0-sdi1 {
 				rockchip,pins =
-					<2 RK_PB6 RK_FUNC_1 &pcfg_pull_none>;
+					<2 RK_PB6 1 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_0_sdi2: i2s-8ch-0-sdi2 {
 				rockchip,pins =
-					<2 RK_PB7 RK_FUNC_1 &pcfg_pull_none>;
+					<2 RK_PB7 1 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_0_sdi3: i2s-8ch-0-sdi3 {
 				rockchip,pins =
-					<2 RK_PC0 RK_FUNC_1 &pcfg_pull_none>;
+					<2 RK_PC0 1 &pcfg_pull_none>;
 			};
 		};
 
 		i2s_8ch_1_m0 {
 			i2s_8ch_1_m0_mclk: i2s-8ch-1-m0-mclk {
 				rockchip,pins =
-					<1 RK_PA2 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PA2 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m0_sclktx: i2s-8ch-1-m0-sclktx {
 				rockchip,pins =
-					<1 RK_PA3 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PA3 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m0_sclkrx: i2s-8ch-1-m0-sclkrx {
 				rockchip,pins =
-					<1 RK_PA4 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PA4 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m0_lrcktx: i2s-8ch-1-m0-lrcktx {
 				rockchip,pins =
-					<1 RK_PA5 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PA5 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m0_lrckrx: i2s-8ch-1-m0-lrckrx {
 				rockchip,pins =
-					<1 RK_PA6 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PA6 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m0_sdo0: i2s-8ch-1-m0-sdo0 {
 				rockchip,pins =
-					<1 RK_PA7 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PA7 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m0_sdo1_sdi3: i2s-8ch-1-m0-sdo1-sdi3 {
 				rockchip,pins =
-					<1 RK_PB0 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PB0 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m0_sdo2_sdi2: i2s-8ch-1-m0-sdo2-sdi2 {
 				rockchip,pins =
-					<1 RK_PB1 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PB1 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m0_sdo3_sdi1: i2s-8ch-1-m0-sdo3_sdi1 {
 				rockchip,pins =
-					<1 RK_PB2 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PB2 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m0_sdi0: i2s-8ch-1-m0-sdi0 {
 				rockchip,pins =
-					<1 RK_PB3 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PB3 2 &pcfg_pull_none>;
 			};
 		};
 
 		i2s_8ch_1_m1 {
 			i2s_8ch_1_m1_mclk: i2s-8ch-1-m1-mclk {
 				rockchip,pins =
-					<1 RK_PB4 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PB4 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m1_sclktx: i2s-8ch-1-m1-sclktx {
 				rockchip,pins =
-					<1 RK_PB5 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PB5 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m1_sclkrx: i2s-8ch-1-m1-sclkrx {
 				rockchip,pins =
-					<1 RK_PB6 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PB6 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m1_lrcktx: i2s-8ch-1-m1-lrcktx {
 				rockchip,pins =
-					<1 RK_PB7 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PB7 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m1_lrckrx: i2s-8ch-1-m1-lrckrx {
 				rockchip,pins =
-					<1 RK_PC0 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PC0 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m1_sdo0: i2s-8ch-1-m1-sdo0 {
 				rockchip,pins =
-					<1 RK_PC1 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PC1 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m1_sdo1_sdi3: i2s-8ch-1-m1-sdo1-sdi3 {
 				rockchip,pins =
-					<1 RK_PC2 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PC2 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m1_sdo2_sdi2: i2s-8ch-1-m1-sdo2-sdi2 {
 				rockchip,pins =
-					<1 RK_PC3 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PC3 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m1_sdo3_sdi1: i2s-8ch-1-m1-sdo3_sdi1 {
 				rockchip,pins =
-					<1 RK_PC4 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PC4 2 &pcfg_pull_none>;
 			};
 
 			i2s_8ch_1_m1_sdi0: i2s-8ch-1-m1-sdi0 {
 				rockchip,pins =
-					<1 RK_PC5 RK_FUNC_2 &pcfg_pull_none>;
+					<1 RK_PC5 2 &pcfg_pull_none>;
 			};
 		};
 
@@ -1714,181 +1714,181 @@
 			lcdc_ctl: lcdc-ctl {
 				rockchip,pins =
 					/* dclk */
-					<1 RK_PA0 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PA0 1 &pcfg_pull_none>,
 					/* hsync */
-					<1 RK_PA1 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PA1 1 &pcfg_pull_none>,
 					/* vsync */
-					<1 RK_PA2 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PA2 1 &pcfg_pull_none>,
 					/* den */
-					<1 RK_PA3 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PA3 1 &pcfg_pull_none>,
 					/* d0 */
-					<1 RK_PA4 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PA4 1 &pcfg_pull_none>,
 					/* d1 */
-					<1 RK_PA5 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PA5 1 &pcfg_pull_none>,
 					/* d2 */
-					<1 RK_PA6 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PA6 1 &pcfg_pull_none>,
 					/* d3 */
-					<1 RK_PA7 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PA7 1 &pcfg_pull_none>,
 					/* d4 */
-					<1 RK_PB0 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PB0 1 &pcfg_pull_none>,
 					/* d5 */
-					<1 RK_PB1 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PB1 1 &pcfg_pull_none>,
 					/* d6 */
-					<1 RK_PB2 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PB2 1 &pcfg_pull_none>,
 					/* d7 */
-					<1 RK_PB3 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PB3 1 &pcfg_pull_none>,
 					/* d8 */
-					<1 RK_PB4 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PB4 1 &pcfg_pull_none>,
 					/* d9 */
-					<1 RK_PB5 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PB5 1 &pcfg_pull_none>,
 					/* d10 */
-					<1 RK_PB6 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PB6 1 &pcfg_pull_none>,
 					/* d11 */
-					<1 RK_PB7 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PB7 1 &pcfg_pull_none>,
 					/* d12 */
-					<1 RK_PC0 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PC0 1 &pcfg_pull_none>,
 					/* d13 */
-					<1 RK_PC1 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PC1 1 &pcfg_pull_none>,
 					/* d14 */
-					<1 RK_PC2 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PC2 1 &pcfg_pull_none>,
 					/* d15 */
-					<1 RK_PC3 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PC3 1 &pcfg_pull_none>,
 					/* d16 */
-					<1 RK_PC4 RK_FUNC_1 &pcfg_pull_none>,
+					<1 RK_PC4 1 &pcfg_pull_none>,
 					/* d17 */
-					<1 RK_PC5 RK_FUNC_1 &pcfg_pull_none>;
+					<1 RK_PC5 1 &pcfg_pull_none>;
 			};
 
 			lcdc_rgb888_m0: lcdc-rgb888-m0 {
 				rockchip,pins =
 					/* d18 */
-					<1 RK_PC6 RK_FUNC_6 &pcfg_pull_none>,
+					<1 RK_PC6 6 &pcfg_pull_none>,
 					/* d19 */
-					<1 RK_PC7 RK_FUNC_6 &pcfg_pull_none>,
+					<1 RK_PC7 6 &pcfg_pull_none>,
 					/* d20 */
-					<2 RK_PB1 RK_FUNC_3 &pcfg_pull_none>,
+					<2 RK_PB1 3 &pcfg_pull_none>,
 					/* d21 */
-					<2 RK_PB2 RK_FUNC_3 &pcfg_pull_none>,
+					<2 RK_PB2 3 &pcfg_pull_none>,
 					/* d22 */
-					<2 RK_PB7 RK_FUNC_3 &pcfg_pull_none>,
+					<2 RK_PB7 3 &pcfg_pull_none>,
 					/* d23 */
-					<2 RK_PC0 RK_FUNC_3 &pcfg_pull_none>;
+					<2 RK_PC0 3 &pcfg_pull_none>;
 			};
 
 			lcdc_rgb888_m1: lcdc-rgb888-m1 {
 				rockchip,pins =
 					/* d18 */
-					<3 RK_PA6 RK_FUNC_3 &pcfg_pull_none>,
+					<3 RK_PA6 3 &pcfg_pull_none>,
 					/* d19 */
-					<3 RK_PA7 RK_FUNC_3 &pcfg_pull_none>,
+					<3 RK_PA7 3 &pcfg_pull_none>,
 					/* d20 */
-					<3 RK_PB0 RK_FUNC_3 &pcfg_pull_none>,
+					<3 RK_PB0 3 &pcfg_pull_none>,
 					/* d21 */
-					<3 RK_PB1 RK_FUNC_3 &pcfg_pull_none>,
+					<3 RK_PB1 3 &pcfg_pull_none>,
 					/* d22 */
-					<3 RK_PB2 RK_FUNC_4 &pcfg_pull_none>,
+					<3 RK_PB2 4 &pcfg_pull_none>,
 					/* d23 */
-					<3 RK_PB3 RK_FUNC_4 &pcfg_pull_none>;
+					<3 RK_PB3 4 &pcfg_pull_none>;
 			};
 		};
 
 		pdm_m0 {
 			pdm_m0_clk: pdm-m0-clk {
 				rockchip,pins =
-					<1 RK_PA4 RK_FUNC_3 &pcfg_pull_none>;
+					<1 RK_PA4 3 &pcfg_pull_none>;
 			};
 
 			pdm_m0_sdi0: pdm-m0-sdi0 {
 				rockchip,pins =
-					<1 RK_PB3 RK_FUNC_3 &pcfg_pull_none>;
+					<1 RK_PB3 3 &pcfg_pull_none>;
 			};
 
 			pdm_m0_sdi1: pdm-m0-sdi1 {
 				rockchip,pins =
-					<1 RK_PB2 RK_FUNC_3 &pcfg_pull_none>;
+					<1 RK_PB2 3 &pcfg_pull_none>;
 			};
 
 			pdm_m0_sdi2: pdm-m0-sdi2 {
 				rockchip,pins =
-					<1 RK_PB1 RK_FUNC_3 &pcfg_pull_none>;
+					<1 RK_PB1 3 &pcfg_pull_none>;
 			};
 
 			pdm_m0_sdi3: pdm-m0-sdi3 {
 				rockchip,pins =
-					<1 RK_PB0 RK_FUNC_3 &pcfg_pull_none>;
+					<1 RK_PB0 3 &pcfg_pull_none>;
 			};
 		};
 
 		pdm_m1 {
 			pdm_m1_clk: pdm-m1-clk {
 				rockchip,pins =
-					<1 RK_PB6 RK_FUNC_4 &pcfg_pull_none>;
+					<1 RK_PB6 4 &pcfg_pull_none>;
 			};
 
 			pdm_m1_sdi0: pdm-m1-sdi0 {
 				rockchip,pins =
-					<1 RK_PC5 RK_FUNC_4 &pcfg_pull_none>;
+					<1 RK_PC5 4 &pcfg_pull_none>;
 			};
 
 			pdm_m1_sdi1: pdm-m1-sdi1 {
 				rockchip,pins =
-					<1 RK_PC4 RK_FUNC_4 &pcfg_pull_none>;
+					<1 RK_PC4 4 &pcfg_pull_none>;
 			};
 
 			pdm_m1_sdi2: pdm-m1-sdi2 {
 				rockchip,pins =
-					<1 RK_PC3 RK_FUNC_4 &pcfg_pull_none>;
+					<1 RK_PC3 4 &pcfg_pull_none>;
 			};
 
 			pdm_m1_sdi3: pdm-m1-sdi3 {
 				rockchip,pins =
-					<1 RK_PC2 RK_FUNC_4 &pcfg_pull_none>;
+					<1 RK_PC2 4 &pcfg_pull_none>;
 			};
 		};
 
 		pdm_m2 {
 			pdm_m2_clkm: pdm-m2-clkm {
 				rockchip,pins =
-					<2 RK_PA4 RK_FUNC_3 &pcfg_pull_none>;
+					<2 RK_PA4 3 &pcfg_pull_none>;
 			};
 
 			pdm_m2_clk: pdm-m2-clk {
 				rockchip,pins =
-					<2 RK_PA6 RK_FUNC_2 &pcfg_pull_none>;
+					<2 RK_PA6 2 &pcfg_pull_none>;
 			};
 
 			pdm_m2_sdi0: pdm-m2-sdi0 {
 				rockchip,pins =
-					<2 RK_PB5 RK_FUNC_2 &pcfg_pull_none>;
+					<2 RK_PB5 2 &pcfg_pull_none>;
 			};
 
 			pdm_m2_sdi1: pdm-m2-sdi1 {
 				rockchip,pins =
-					<2 RK_PB6 RK_FUNC_2 &pcfg_pull_none>;
+					<2 RK_PB6 2 &pcfg_pull_none>;
 			};
 
 			pdm_m2_sdi2: pdm-m2-sdi2 {
 				rockchip,pins =
-					<2 RK_PB7 RK_FUNC_2 &pcfg_pull_none>;
+					<2 RK_PB7 2 &pcfg_pull_none>;
 			};
 
 			pdm_m2_sdi3: pdm-m2-sdi3 {
 				rockchip,pins =
-					<2 RK_PC0 RK_FUNC_2 &pcfg_pull_none>;
+					<2 RK_PC0 2 &pcfg_pull_none>;
 			};
 		};
 
 		spdif_in {
 			spdif_in: spdif-in {
 				rockchip,pins =
-					<0 RK_PC2 RK_FUNC_1 &pcfg_pull_none>;
+					<0 RK_PC2 1 &pcfg_pull_none>;
 			};
 		};
 
 		spdif_out {
 			spdif_out: spdif-out {
 				rockchip,pins =
-					<0 RK_PC1 RK_FUNC_1 &pcfg_pull_none>;
+					<0 RK_PC1 1 &pcfg_pull_none>;
 			};
 		};
 
@@ -1923,7 +1923,7 @@
 
 			uart0_rts_gpio: uart0-rts-gpio {
 				rockchip,pins =
-					<2 RK_PA3 RK_FUNC_GPIO &pcfg_pull_none>;
+					<2 RK_PA3 0 &pcfg_pull_none>;
 			};
 		};
 
@@ -1998,7 +1998,7 @@
 
 			uart4_rts_gpio: uart4-rts-gpio {
 				rockchip,pins =
-					<4 RK_PA7 RK_FUNC_GPIO &pcfg_pull_none>;
+					<4 RK_PA7 0 &pcfg_pull_none>;
 			};
 		};
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3308b-evb-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3308b-evb-v10.dtsi
index 36d66e622134..4f61c53bde70 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308b-evb-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308b-evb-v10.dtsi
@@ -676,25 +676,25 @@
 
 	buttons {
 		pwr_key: pwr-key {
-			rockchip,pins = <0 RK_PA6 RK_FUNC_GPIO &pcfg_pull_up>;
+			rockchip,pins = <0 RK_PA6 0 &pcfg_pull_up>;
 		};
 	};
 
 	usb {
 		usb_drv: usb-drv {
-			rockchip,pins = <0 RK_PC5 RK_FUNC_GPIO &pcfg_pull_none>;
+			rockchip,pins = <0 RK_PC5 0 &pcfg_pull_none>;
 		};
 	};
 
 	sdio-pwrseq {
 		wifi_enable_h: wifi-enable-h {
-			rockchip,pins = <4 RK_PD6 RK_FUNC_GPIO &pcfg_pull_none>;
+			rockchip,pins = <4 RK_PD6 0 &pcfg_pull_none>;
 		};
 	};
 
 	wireless-wlan {
 		wifi_wake_host: wifi-wake-host {
-			rockchip,pins = <2 RK_PB4 RK_FUNC_GPIO &pcfg_pull_up>;
+			rockchip,pins = <2 RK_PB4 0 &pcfg_pull_up>;
 		};
 	};
 };
-- 
2.35.3

